code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -90,6 +91,28 @@
 
 	return dat
 
+	/mob/living/simple_animal/bot/floorbot/attackby(obj/item/W , mob/user, params)
+		if(istype(W, /obj/item/stack/tile/plasteel))
+			user << "<span class='notice'>The floorbot can produce normal tiles itself.</span>"
+			return
+		if(specialtiles != 0 && istype(W, /obj/item/stack/tile/)
+			var/obj/item/stack/tile/usedtile = W
+			if(usedtile != tiletype)
+				user << "<span class='warning'>Different custom tiles are already inside the floorbot.</span>"
+				return
+		if(istype(W, /obj/item/stack/tile/) && specialtiles == 0)
+			tiletype = W
+		if(specialtiles >= 100)
+			return
+		var/loaded = min(100-specialtiles, tiletype.amount)
+		tiletype.use(loaded)
+		specialtiles += loaded
+		if(loaded > 0)
+			user << "<span class='notice'>You load [loaded] tiles into the floorbot. He now contains [specialtiles] tiles.</span>"
```

Comment:
Oh, that was one of the lines I just copied from the old code. It's probably been there for years. I'll fix that too.